README.md

```diff
@@ -81,6 +81,16 @@ Specific versions: `1.15`, `1.16.1`, `1.17.0-rc2`, `1.16.0-beta1`
 SemVer's version range syntax: `^1.13.1`  
 For more information about semantic versioning please refer [semver](https://github.com/npm/node-semver) documentation
 
+
+# Keep in mind: latest, cached go compilers and semver notation.
+
+The `setup-go` action doesn't install the latest matched version if the cached version matches one from WF file.
+
+**For example:**  
+Currently, there is three go compilers on [Ubuntu 20.04.3 LTS](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md) virtual environment: `1.15.15`, `1.16.12`, `1.17.5`.  
+When the `1.17.6` version will be out and there is `1.17.*` in `go-version` field, the cached `1.17.5` version will be used. Not the new one. Until the cached version will be updated to the latest `1.17.6`.   
```

Comment:
> When the `1.17.6` version will be out and there is `1.17.*` in `go-version` field, the cached `1.17.5` version will be used. Not the new one. Until the cached version will be updated to the latest `1.17.6`. 

Users also want to know what should be in the `go-version` field if they want the latest 1.17 to always be downloaded when a newer 1.17 version than the cached version becomes available.  When 1.17.6 is released, it will download that without any modifications to GitHub Actions workflow.  When 1.17.7 is released, it will download that without any modifications to GitHub Actions workflow.  And etc.

It would be helpful to users if text using a simple example is added to answer this.